proposals/0111-linear-types.rst

```diff
@@ -907,6 +909,159 @@ or ``case`` in the surface syntax), we want to infer the multiplicity
 annotation. The process for this is not yet defined (see `Unresolved
 questions`_ below for a more precise description of this issue).
 
+Let and where bindings
+----------------------
+
+*This section is written with let bindings, but all of the same applies
+to where bindings.*
+
+Note on terminology: following the Haskell 2010 report, a function
+binding is a binding of the form ``f arg1…argn =`` with at least one
+arguments. The other binding is called pattern bindings. In particular
+``let x =`` is a pattern binding (as opposed to how it is in GHC's
+implementation, where ``let x =`` would be a ``FunBind``). This
+proposal will be using the terminology “non-variable pattern”
+for patterns which aren't a single variable.
+
+Let bindings can optionally be annotated with a multiplicity
+(including a multiplicity variable):
+
+::
+
+   let
+     %p1 pat1 = rhs1
+     …
+     %pn patn = rhsn
+   in body
+
+if
+
+* The binding is not top-level
+* The binding is non-recursive
```

Comment:
But I think

```
let %1 x = ... y ...
    y = ... x ...
```

is probably OK. But maybe this can't happen, because `x` cannot occur in the RHS of the unrestricted `y`. So perhaps I'm just making noise here. Resolve after reading.